File: python/src/anni/analyser.py
from typing import List
from anni.types.datatypes import Person, ScoredPerson
from anni.calculator.calculator import AutomatedNaughtyOrNiceCalculator
from anni.requests.requests import RequestHandler

class AutomatedNaughtyOrNiceAnalyser:
    def __init__(self, calculator: AutomatedNaughtyOrNiceCalculator | None = None, handler: RequestHandler | None = None):
        self.calculator = calculator or AutomatedNaughtyOrNiceCalculator()
        self.handler = handler or RequestHandler()

    def analyse(self, people: List[Person]) -> List[ScoredPerson]:
        return self.calculator.classify_people(people)

    def submit(self, scored: List[ScoredPerson]):
        return self.handler.submit_list(scored)

File: javascript/src/anni/analyser.js
import { AutomatedNaughtyOrNiceCalculator } from './calculator/calculator.js';
import { RequestHandler } from './requests/requests.js';

export class AutomatedNaughtyOrNiceAnalyser {
  constructor(calculator = null, handler = null) {
    this.calculator = calculator || new AutomatedNaughtyOrNiceCalculator();
    this.handler = handler || new RequestHandler();
  }

  analyse(people) {
    return this.calculator.classifyPeople(people);
  }

  async submit(scoredPeople) {
    return await this.handler.submitList(scoredPeople);
  }
}


